openff/toolkit/topology/molecule.py

```diff
@@ -2444,6 +2444,26 @@ def assign_partial_charges(
         openff.toolkit.utils.toolkits.AmberToolsToolkitWrapper.assign_partial_charges
         openff.toolkit.utils.toolkits.BuiltInToolkitWrapper.assign_partial_charges
         """
+
+        # Raise a warning when users try to apply these charge methods to "large" molecules
+        WARN_LARGE_MOLECULES: Set[str] = {
+            "am1bcc",
+            "am1bccelf10",
+            "am1-mulliken",
+            "am1bccnosymspt",
+            "am1elf10",
+        }
+
+        if partial_charge_method in WARN_LARGE_MOLECULES:
+            if self.n_atoms > 150:
+                warnings.warn(
+                    f"Warning! Partial charge method '{partial_charge_method}' is not designed "
+                    "for use on large (i.e. > 150 atoms) molecules and may crash or take hours to "
+                    f"run on this molecule (found {self.n_atoms} atoms). For more, see "
+                    "https://docs.openforcefield.org/projects/toolkit/en/topology-biopolymer-refactor/faq.html"
+                    "#parameterizing-my-system-which-contains-a-large-molecule-is-taking-forever-whats-wrong",
+                )
```

Comment:
It would be nice to add a section covering alternatives, like
* If you already know your charges, use `charge_from molecules` ...
* If your charges fit a pattern, here are some instructions on how to write up a `LibraryChargeType` ...

but I didn't want to write too much here. Maybe it's worth having a paragraph on that in this FAQ entry.